### ISSUE TITLE
request does not properly parse body, when transfer-encoding is set to "chunked"

### ISSUE THREAD

**Issue by pano9000** [nodejs/undici#2421]
## Bug Description

When working with REST APIs that return a larger number of data and use 'transfer-encoding': 'chunked', `request` is not able to properly parse the body with the json() or text() method.
The text() method will work, but it will cut off the received data, e.g. in the middle of the data, if you inspect the text.
The json() method throws SyntaxError, because the body is of course not well-formed due to the cut off - the exact error message will differ, because the data is cut off at different lengths, e.g. on the same API, I might get the following errors.

`Uncaught SyntaxError: Exponent part is missing a number in JSON at position 68576`
`Uncaught SyntaxError: Expected ',' or '}' after property value in JSON at position 68575`
`Uncaught SyntaxError: Expected double-quoted property name in JSON at position 68579`

Occasionally it might happen that it actually suceeds to properly parse the body.

Interestingly enough, when using `fetch`, this issue does not occur at all, as far as I saw.
Am I potentially missing something here?

## Reproducible By

Some example fake APIs with large enough datasets, that use chunked transfer-encoding, which can be used to reproduce this
https://fakerapi.it/api/v1/credit_cards?_quantity=2000
https://jsonplaceholder.typicode.com/comments

```
const response = await undici.request("https://jsonplaceholder.typicode.com/comments");
const data = await response.body.json();
```

## Expected Behavior

It should correctly return the data in a reproducible way

## Environment

Node 20, 
undici 5.27.2

**Comment by Uzlopak** [comment]
I can not reproduce this. 

I used node 20.0.0 and 20.9.0. And also used undici 5.27.2.

**Comment by pano9000** [comment]
Hi, thanks for the comment-
It does look like the example APIs above may have adjusted the number of entries they are outputting, e.g. the fakerapi is now limiting the data to 1000 entries.

I will do some further tests this evening when I get back home and report back.

**Comment by Uzlopak** [comment]
Please do. I am really interested ;)

**Comment by pano9000** [comment]
ok, did some testing and even resorted to creating a local rest API that sends data with transfer-encoding set to chunk.
I even tested with as high as 400k entries in an array - and I was not able to trigger the error anymore, oddly enough.
Tested with Node 18 and Node 20.11 and undici 5.27.2.

Let me do one more test tomorrow at my workplace, with a non-private API we are running, which is where I first encountered this behaviour with - if that also works fine I'll go ahead and close this bug.

Will report back tomorrow again!
